# Stefan Teleman <stefan.teleman@oracle.com>
# Make sure this struct is always aligned on 8.
# Internal patch. Submitted upstream. Status pending.
--- libgomp/config/posix/ptrlock.h	2014-01-02 17:25:54.000000000 -0500
+++ libgomp/config/posix/ptrlock.h	2014-11-19 08:35:00.554310487 -0500
@@ -28,7 +28,10 @@
 #ifndef GOMP_PTRLOCK_H
 #define GOMP_PTRLOCK_H 1
 
-typedef struct { void *ptr; gomp_mutex_t lock; } gomp_ptrlock_t;
+typedef struct {
+  void *ptr;
+  gomp_mutex_t lock;
+} gomp_ptrlock_t __attribute__((__aligned__(8)));
 
 static inline void gomp_ptrlock_init (gomp_ptrlock_t *ptrlock, void *ptr)
 {
